File: workvpn-ios/WorkVPN/Models/VPNConfig.swift
//
//  VPNConfig.swift
//  WorkVPN
//
//  Model for VPN configuration
//

import Foundation

struct VPNConfig: Codable {
    var name: String
    var content: String
    var serverAddress: String
    var port: Int
    var `protocol`: String
    var importedAt: Date

    // Parsed certificate components
    var ca: String?
    var cert: String?
    var key: String?
    var tlsAuth: String?
    var cipher: String?
    var auth: String?

    init(name: String, content: String, serverAddress: String, port: Int = 1194, protocol: String = "udp") {
        self.name = name
        self.content = content
        self.serverAddress = serverAddress
        self.port = port
        self.protocol = `protocol`
        self.importedAt = Date()
    }
}

extension VPNConfig {
    var displayName: String {
        name.replacingOccurrences(of: ".ovpn", with: "")
    }

    var protocolDisplay: String {
        "\(`protocol`.uppercased()):\(port)"
    }
}

File: workvpn-android/app/src/main/java/com/workvpn/android/model/VPNConfig.kt
package com.barqnet.android.model

import kotlinx.serialization.Serializable
import java.util.Date

@Serializable
data class VPNConfig(
    val name: String,
    val content: String,
    val serverAddress: String,
    val port: Int = 1194,
    val protocol: String = "udp",
    val importedAt: Long = System.currentTimeMillis(),

    // Parsed certificate components
    val ca: String? = null,
    val cert: String? = null,
    val key: String? = null,
    val tlsAuth: String? = null,
    val cipher: String? = null,
    val auth: String? = null,
    val deviceType: String = "tun"
) {
    val displayName: String
        get() = name.removeSuffix(".ovpn")

    val protocolDisplay: String
        get() = "${protocol.uppercase()}:$port"

    val importedDate: Date
        get() = Date(importedAt)
}
